### ISSUE TITLE
Deserializing .sav file

### ISSUE THREAD

**Issue by jayjayb772** [gibbed/Gibbed.Borderlands3#3]
I've been trying to figure out how to decode the .sav file from BL3 but still to no avail. Have you been able to find the the Hash algorithm to properly deserialize these .sav files?

**Comment by NicoDblc** [comment]
Have you tried using [Gibbed.Borderlands3.ProtoBufFormats](https://github.com/gibbed/Gibbed.Borderlands3/tree/master/projects/Gibbed.Borderlands3.ProtoBufFormats) ?

**Comment by IOnlyFetchBranches** [comment]
aren't those just protobuf formats like data contracts or something? not seeing how that helps decode a. sav file. But my only protobuff experience comes from gRpc 😂

**Comment by denverquane** [comment]
I've been messing with the data today, and I'm fairly certain that while the underlying data is protobuf, it seems to be obfuscated or encrypted with something else. I couldn't unmarshal any .sav files with the .proto definitions, and from Gibbed's Twitter it seems like there's something else wrapping the raw data.

**Comment by NicoDblc** [comment]
Yes that's exactly the experience I had as well, trying to deserialize the data, using `CharacterData` gives an error which leads to that conclusion

**Comment by denverquane** [comment]
> trying to deserialize the data, using `CharacterData` gives an error

Yeah same with 'Character', 'Profile', etc in my case. He said on Twitter something about the .Pak files being encrypted with AES, but I'm not sure if the save files are the same (I saw two labeled "blocks" in the data with textual labels, seems unlikely)

**Comment by ensiform** [comment]
Maybe best to wait and see an update from him when he has some time?
